File: models/pharmacy/StockAdjustment.js
import mongoose from "mongoose";

const { Schema } = mongoose;

const StockAdjustmentSchema = new Schema(
  {
    hospitalId: {
      type: Schema.Types.ObjectId,
      ref: "Hospital",
      required: true
    },

    branchId: {
      type: Schema.Types.ObjectId,
      ref: "Branch",
      required: true
    },

    stockId: {
      type: Schema.Types.ObjectId,
      ref: "PharmacyStock",
      required: true
    },

    adjustmentType: {
      type: String,
      enum: ["Damage", "Expired", "AuditCorrection"],
      required: true
    },

    quantity: {
      type: Number,
      required: true
    },

    reason: {
      type: String,
      trim: true
    },

    createdBy: {
      type: Schema.Types.ObjectId,
      ref: "User",
      required: true
    },

    approvedBy: {
      type: Schema.Types.ObjectId,
      ref: "User"
    }
  },
  { timestamps: true }
);

export default mongoose.model("StockAdjustment", StockAdjustmentSchema);

File: jstojv/src/main/java/com/samrat/jstojv/entity/StockAdjustment.java
package com.samrat.jstojv.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "stock_adjustments")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class StockAdjustment extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "hospital_id", nullable = false)
    private Hospital hospital;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "branch_id", nullable = false)
    private Branch branch;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "stock_id", nullable = false)
    private PharmacyStock stock;

    @Enumerated(EnumType.STRING)
    @Column(name = "adjustment_type", nullable = false)
    private AdjustmentType adjustmentType;

    @Column(nullable = false)
    private Integer quantity;

    private String reason;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by", nullable = false)
    private User createdBy;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "approved_by")
    private User approvedBy;

    public enum AdjustmentType {
        Damage, Expired, AuditCorrection
    }
}
